RELEASE/scripts/autoscend/paths/avatar_of_shadows_over_loathing.ash

```diff
@@ -0,0 +1,590 @@
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=16", true);
+				set_property("auto_aosolLastSkill", 6);
+			}
+			if(!have_skill($skill[Sax of Violence]) && my_level()>=7 && my_meat()>700 + meatReserve()) //Deal Mus in Sleaze dmg
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=17", true);
+				set_property("auto_aosolLastSkill", 6);
+			}
+			if(!have_skill($skill[Rhythmic Precision]) && my_level()>=7 && my_meat()>700 + meatReserve()) //-3 MP to use skills
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=7", true);
+				set_property("auto_aosolLastSkill", 7);
+			}
+			if(!have_skill($skill[Tricky Timpani]) && my_level()>=8 && my_meat()>800 + meatReserve()) //Tricky Timpani (10 advs, +5 prismatic dmg)
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=18", true);
+				set_property("auto_aosolLastSkill", 7);
+			}
+			if(!have_skill($skill[Perfect Embouchure]) && my_level()>=8 && my_meat()>800 + meatReserve()) //Musical skills deal 33% more damage
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=8", true);
+				set_property("auto_aosolLastSkill", 8);
+			}
+			if(!have_skill($skill[Grit Teeth]) && my_level()>=9 && my_meat()>900 + meatReserve()) //In combat 20 HP heal
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=19", true);
+				set_property("auto_aosolLastSkill", 8);
+			}
+			if(!have_skill($skill[Improv Muscles]) && my_level()>=9 && my_meat()>900 + meatReserve()) //+25% Max HP, +25% Initiatve
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=9", true);
+				set_property("auto_aosolLastSkill", 9);
+			}
+			if(!have_skill($skill[Impeccable Timing]) && my_level()>=10 && my_meat()>1000 + meatReserve()) //passive +100% combat initiative
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=2", true);
+				set_property("auto_aosolLastSkill", 9);
+			}
+			if(!have_skill($skill[Soothing Flute]) && my_level()>=10 && my_meat()>1000 + meatReserve()) //Soothing Flute (10 advs, +5 fam weight, regen 8-10 hp per adv)
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=20", true);
+				set_property("auto_aosolLastSkill", 10);
+			}
+			if(!have_skill($skill[Rhythm In Your Blood]) && my_level()>=11 && my_meat()>1100 + meatReserve()) //+20% Max HP
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=11", true);
+				set_property("auto_aosolLastSkill", 10);
+			}
+			if(!have_skill($skill[Motif]) && my_level()>=11 && my_meat()>1100 + meatReserve()) //25 turn blue ray (olfaction-esque)
+			{
+				page = visit_url("choice.php?pwd&whichchoice=1495&option=1&whichsk=21", true);
+				set_property("auto_aosolLastSkill", 100);
+			}
+		}
+	}
+	else
+	{
+		return false;
+	}
+	
+	return false;
+}
+
+boolean auto_pigSkinnerAcquireHP(int goal)
+{
+	while (my_hp() < goal)
```

Comment:
Not understanding point of these while loops. What am I missing?

Reply:
They cause auto_restore to break which allows us to ignore buying/using healing items and use the healing combat skills instead. I'm actually not convinced that I have them actually written correctly, but they do what I intended so I guess if it ain't broke?

They are only called if the player has the healing combat skills for each class.

Reply:
Interesting! Not familiar enough with the restore code to know if there is a more elegant way to do this. I'm fine with keeping code as is but please add a comment about what they do